pubsub/gcppubsub/gcppubsub.go

```diff
@@ -387,19 +407,34 @@ func (*topic) ErrorCode(err error) gcerrors.ErrorCode {
 func (*topic) Close() error { return nil }
 
 type subscription struct {
-	client *raw.SubscriberClient
-	path   string
+	client  *raw.SubscriberClient
+	path    string
+	options *SubscriptionOptions
 }
 
 // SubscriptionOptions will contain configuration for subscriptions.
-type SubscriptionOptions struct{}
+type SubscriptionOptions struct {
+	// MaxBatchSize caps the maximum batch sized used when retrieving messages. It defaults to 1000.
+	MaxBatchSize int
+}
 
 // OpenSubscription returns a *pubsub.Subscription backed by an existing GCP
 // PubSub subscription subscriptionName in the given projectID. See the package
 // documentation for an example.
 func OpenSubscription(client *raw.SubscriberClient, projectID gcp.ProjectID, subscriptionName string, opts *SubscriptionOptions) *pubsub.Subscription {
 	path := fmt.Sprintf("projects/%s/subscriptions/%s", projectID, subscriptionName)
-	return pubsub.NewSubscription(openSubscription(client, path), nil, ackBatcherOpts)
+
+	if opts == nil {
```

Comment:
To make sure we're always getting the default as documented, and to share code better between the two constructors, how about:

```
dsub := openSubscription(client, path, opts)
recvOpts := *defaultRecvBatcherOpts
recvOpts.MaxBatchSize = dsub.opts.MaxBatchSize
return pubsub.NewSubscription(dsub, recvOpts, ackBatcherOpts)
```

here and in `OpenSubscriptionByPath`.

Then in `openSubscription`:

```
if opts == nil {
  opts = &SubscriptionOptions{}
}
if opts.MaxBatchSize == 0 {
  opts.MaxBatchSize = defaultRecvBatchOpts.MaxBatchSize
}
return &subscription{client, subscriptionPath, opts}
```

I think for that to work you'll need to change the return value for `openSubscription` to `*subscription`, but that should be fine.
?

Reply:
Yep, this makes sense to me